frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -2,26 +2,56 @@ import React, { useState } from 'react';
 import { useHistory } from 'react-router-dom';
 import styled from 'styled-components';
 import MarkdownEditor from 'rich-markdown-editor';
-import { deleteProblem, Problem, ProblemIOType } from '../../api/Problem';
 import {
-  ConsoleTextArea, LargeInputButton, TextInput, CheckboxInput,
+  deleteProblem,
+  Problem,
+  ProblemIOType,
+  problemIOTypeToString,
+} from '../../api/Problem';
+import {
+  FixedTextArea,
+  PureTextInputTitle,
+  TextInput,
+  CheckboxInput,
 } from '../core/Input';
-import Difficulty from '../../api/Difficulty';
-import { DifficultyButton, ProblemIOTypeButton, SmallButton } from '../core/Button';
-import { MediumText, Text } from '../core/Text';
+import { Difficulty } from '../../api/Difficulty';
+import {
+  SmallDifficultyButton,
+  PrimaryButton,
+  TextButton,
+  SmallButton,
+} from '../core/Button';
+import PrimarySelect from '../core/Select';
+import { SmallHeaderText, LowMarginMediumText, Text } from '../core/Text';
 import Loading from '../core/Loading';
 import ErrorMessage from '../core/Error';
+import { ThemeConfig } from '../config/Theme';
+import { SmallButtonLink } from '../core/Link';
+import { FlexBareContainer } from '../core/Container';
+
+const MainContent = styled.div`
+  text-align: left;
+  padding: 20px;
+  flex: 6;
+`;
 
-const StyledMarkdownEditor = styled(MarkdownEditor)`
+const SidebarContent = styled.div`
   text-align: left;
-  padding: 35px;
-  border-radius: 5px;
+  padding: 20px;
+  flex: 4;
+`;
+
+const SettingsContainer = styled.div<any>`
+  text-align: left;
+  margin: 0.25rem 0 1rem 0;
+  padding: ${({ padding }) => padding || '1rem'};
+  border-radius: 10px;
   box-shadow: 0 -1px 4px rgba(0, 0, 0, 0.12);
   background: ${({ theme }) => theme.colors.white};
 `;
 
-const Content = styled.div`
-  padding: 10px;
+const MarginRightContainer = styled.div<any>`
+  margin-right: ${({ marginRight }) => marginRight}%;
```

Comment:
nit: Create simple types to replace any

Reply:
I got rid of the styled components that necessitated this as I simply creates new components for this use 👍